### PR TITLE
Sha3 unit test functions.

### PR DESCRIPTION
**Pull request by jrblixt**

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -3958,6 +3967,866 @@ static int test_wc_RipeMdFinal (void)
...
+    #endif /* SHA3_512 */
+#endif /* WOLFSSL_SHA3 */
+    return ret;
+
+} /* END testing_wc_Sha3_Update */
+
+/*
+ *  Testing wc_Sha3_224_Final()
+ */
+static int test_wc_Sha3_224 (void)
```
Comment (cconlon) [inline tests/api.c]: Should this function be named "test_wc_Sha3_224_Final()" to match your naming scheme?

File: tests/api.c [hunk 2]
```diff
@@ -3958,6 +3967,866 @@ static int test_wc_RipeMdFinal (void)
...
+    ret = wc_InitSha3_224(&sha3Cpy, HEAP_HINT, devId);
+    if (ret != 0) {
+    wc_Sha3_224_Free(&sha3);
+    return ret;
+    }
+
+    ret = wc_Sha3_224_Update(&sha3, (byte*)msg, msglen);
+
+    if (ret == 0) {
+        ret = wc_Sha3_224_Copy(&sha3Cpy, &sha3);
```
Comment (cconlon) [inline tests/api.c]: Should we be verifying that the copy works correctly?  For example getting the final hash from both old and new copy and verifying they are the same value?

Same comment applies to other SHA-3 copy functions.

Comment (jrblixt) [inline tests/api.c]: This should show outdated and has been fixed.

**Comment by cconlon** [comment]
If I explicitly disable SHA-3:

```
./configure --disable-sha3
make
```

I get the following error/warning:

```
tests/api.c:167:12: error: unused variable 'devId' [-Werror,-Wunused-variable]
static int devId = INVALID_DEVID;
           ^
1 error generated.
```

Please fix, thanks.

Chrhis

**Pull request merged**